File: backend/app/schemas/document.py
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel


class DocumentBase(BaseModel):
    original_filename: str
    mime_type: str
    file_size: int


class DocumentResponse(DocumentBase):
    id: UUID
    session_id: UUID
    filename: str
    processing_status: str
    processing_error: str | None
    chunk_count: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class DocumentListResponse(BaseModel):
    documents: list[DocumentResponse]
    total: int


class DocumentStatusResponse(BaseModel):
    id: UUID
    processing_status: str
    processing_error: str | None
    chunk_count: int

File: frontend/src/types/document.ts
export interface Document {
  id: string;
  session_id: string;
  filename: string;
  original_filename: string;
  file_size: number;
  mime_type: string;
  processing_status: 'pending' | 'processing' | 'completed' | 'failed';
  processing_error: string | null;
  chunk_count: number;
  created_at: string;
  updated_at: string;
}

export interface DocumentListResponse {
  documents: Document[];
  total: number;
}

export interface DocumentStatusResponse {
  id: string;
  processing_status: string;
  processing_error: string | null;
  chunk_count: number;
}
